File: 2824.count-pairs-whose-sum-is-less-than-target.java
/*
 * @lc app=leetcode id=2824 lang=java
 *
 * [2824] Count Pairs Whose Sum is Less than Target
 */

// @lc code=start

import java.util.Collections;
import java.util.List;

class Solution {
    public int countPairs(List<Integer> nums, int target) {
        if (nums.size() == 1) {
            return 0;
        }
        Collections.sort(nums);
        int left = 0;
        int cnt = 0;
        while (left != nums.size() - 1) {
            int right = left + 1;
            while (right != nums.size() && nums.get(left) + nums.get(right) < target) {
                cnt += 1;
                right += 1;
            }
            left += 1;
        }
        return cnt;
    }
}
// @lc code=end

File: 2824.count-pairs-whose-sum-is-less-than-target.cpp
/*
 * @lc app=leetcode id=2824 lang=cpp
 *
 * [2824] Count Pairs Whose Sum is Less than Target
 */

// @lc code=start
#include <algorithm>
#include <vector>

using namespace std;

class Solution {
public:
    int countPairs(vector<int>& nums, int target) {
        if (nums.size() == 1) return 0;
        sort(nums.begin(), nums.end());
        unsigned cnt = 0;
        for (auto l = nums.cbegin(); l != nums.cend() - 1; ++l) {
            for (auto r = l + 1; r != nums.cend(); ++r) {
                if (*l + *r < target) ++cnt;
            }
        }
        return cnt;
    }
};
// @lc code=end
